File: src/cpp_siro/src/siro_node_publicador.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class SiroNodePublicador : public rclcpp::Node
{
public:
    SiroNodePublicador() : Node("siro_node_publicador")
    {
        // Publica mensajes en el tópico 'chatter' cada segundo
        publisher_ = this->create_publisher<std_msgs::msg::String>("chatter", 10);
        timer_ = this->create_wall_timer(std::chrono::seconds(1),
                                         [this]() {
                                             auto msg = std_msgs::msg::String();
                                             msg.data = "¡Hola desde siro_node_publicador!";
                                             RCLCPP_INFO(this->get_logger(), "Publicando: '%s'", msg.data.c_str());
                                             publisher_->publish(msg);
                                         });
    }

private:
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char *argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SiroNodePublicador>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/python_siro/python_siro/siro_node_publicador.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String
import time

class SiroNodePublicador(Node):

    def __init__(self):
        super().__init__('siro_node_publicador')
        self.publisher_ = self.create_publisher(String, 'chatter', 10)
        self.timer_ = self.create_timer(1, self.timer_callback)

    def timer_callback(self):
        msg = String()
        msg.data = '¡Hola desde siro_node_publicador!'
        self.get_logger().info('Publicando: "%s"' % msg.data)
        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = SiroNodePublicador()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()
